### DISCUSSION TITLE
wire:click - Uncaught TypeError: Cannot read properties of null (reading 'match')

### DISCUSSION THREAD

**Discussion by lucasjose501** [livewire/livewire#4059] [1. Help]
Hello everyone! First of all, thanks for the Livewire project.

I'm building an application that has some dropdowns that sometimes need to get information from the database, but it's not all the time, so I thought about loading these content only when opening the dropdown.

I'm using Livewire and Alpine.js to build it.

To make this work, I've added the code 'wire:click' in the trigger that opens the dropdown to load this information and a conditional check, so that it doesn't run again if the content has already been loaded.
Everything works as expected, the downside is just a console log, so I'm not exactly sure if it's something on livewire or the way I'm using it.

Note that I don't have much experience with livewire and laravel, so maybe I'm doing it wrong.

I created a minimal example repository demonstrating what happens. Details on how to reproduce the problem are in the readme: https://github.com/lucasjose501/livewire-uncaught-typeerror

For the component itself, you can see it on this commit: https://github.com/lucasjose501/livewire-uncaught-typeerror/commit/efc62545a1090061e2be7dfba680ca37cff98711

Thank you for your patience and I sincerely apologize if I am opening a wrong issue.

**Comment by joshhanley** [comment]
@lucasjose501 any chance you could do a small screen capture demonstrating the issue? I use https://gifcap.dev/

Thanks!

**Reply by lucasjose501** [reply]
Not a problem at all!

Here is the gif:
https://github.com/lucasjose501/livewire-uncaught-typeerror/blob/main/public/recording.gif

**Comment by lucasjose501** [comment]
Now that I thought more about the subject and after reading part of the documentation about livewire and alpinejs, I found a way that worked and didn't generate errors, which is alpinejs calling the action, but I believe that in another scenario without alpinejs this would happen again.

I changed it from:
```
<button x-on:click="open = !open" @if(!$loaded) wire:click="load" @endif>
    Dropdown Trigger
</button>
```
to:
```
<button x-on:click="open = !open; @if(!$loaded) $wire.load() @endif">
    Dropdown Trigger
</button>
```

I believe that when the HTML is re-rendered, livewire is not removing the 'on click' event from the dom

**Comment by joshhanley** [comment]
@lucasjose501 yeah I don't think conditionally removing the `wire:click` removes the event listener. What I would probably do is actually `@entangle('loaded')`, that way you could then control the conditional in javascript instead. Like

```blade
<div x-data="{ open: false, loaded: @entangle('loaded') }">
    <button x-on:click="open = !open; if(! loaded) $wire.load()">
        Dropdown Trigger
    </button>
```

Hope this helps!
